raiden/ui/cli.py

```diff
@@ -798,7 +799,10 @@ def append_report(subject, data):
     append_report('raiden version', json.dumps(get_system_spec()))
     append_report('raiden log', None)
 
-    validate_solc()
+    # Binary distributions don't need the solidity compiler and don't include
+    # the smartcontracts code
+    if os.path.exists(get_contract_path('HumanStandardToken.sol')):
```

Comment:
Unfortunately this check can't work because the bundle also includes the contract source in order for the checksum verification to work.

I think using `get_static_or_compile()` with one of the contracts should work as a check.

Reply:
Well, if the bundle includes the contract source then the check should be executed. Or are you saying that:

- The bundle includes the contract source
- *and* the precompiled code
- *and* that solc is **not** required

?

Reply:
Yes to all three :)

Here are the relevant lines:
https://github.com/raiden-network/raiden/blob/3cb5d526f511707d4b4d0fa2840aa60d1cc861c6/raiden/blockchain/abi.py#L102-L104

In order to validate the checksum of the precompiled ABI the contract source has to be available (and the checksum check isn't optional)